### PR TITLE
Implements Comments side bar

### PR DESCRIPTION
**Pull request by MauroGuerrero20**
# Features
- Comments are now displayed in the right sidebar. 
  - If there are no comments, then the right sidebar will not be displayed. 

### Before 
Screenshot 2020-06-04 at 2 05 09 PM

### After
Screenshot 2020-06-04 at 4 23 02 PM

# Demo
Comments SideBar Demo

- ### Note: The webpage is different in the demo because it was recorded after some styling changes.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/containers/Comment.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.google.sps.containers;
+
+public final class Comment{
```
Comment (VivekShankar) [inline portfolio/src/main/java/com/google/sps/containers/Comment.java]: Can you add Javadoc to this class (and the new classes below) describing what they're for?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -73,13 +77,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    List<Comment> commentsList = new ArrayList();
 
     Query commentsQuery = new Query("Comment");
     PreparedQuery results = datastore.prepare(commentsQuery);
 
     for (Entity entity : results.asIterable()){
-      commentsList.add((String)entity.getProperty("comment_msg"));
+
+      Comment cmt = new Comment(
+        (String)entity.getProperty("name"),
```
Comment (VivekShankar) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: add a space between `(String)` and `entity.getProperty`

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -11,157 +11,187 @@
   </head>
   <body>
     <!--Navbar-->
-    <div class="navbar">
+    <div id="navbar" class="navbar navbar-expand-lg">
```
Comment (VivekShankar) [inline portfolio/src/main/webapp/index.html:14]: Looks like there are a lot of line diffs in this file - but I guess it's mostly due to a change in indentation/formatting of the HTML?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -36,6 +36,36 @@ function randQuote() {
...
+
+function addCommentsColumn(){
+
+  document.getElementById("main_body").classList.remove("col-12");
+  document.getElementById("main_body").classList.add("col-10");
+
+  document.getElementById("comments_body").classList.remove("col-0");
+  document.getElementById("comments_body").classList.add("col-2");
+
+  // var comments_div = appendElement("about_me", "div");
```
Comment (VivekShankar) [inline portfolio/src/main/webapp/script.js]: Can you remove the dead code here?

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -36,6 +36,36 @@ function randQuote() {
   }
 }
 
+function appendElement(currentElement, newElement) {
```
Comment (VivekShankar) [inline portfolio/src/main/webapp/script.js]: I like how you've structured the DOM manipulation here into clearly named functions!

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -45,26 +75,45 @@ function getComments() {
...
-    const commentsOutput = document.getElementById("comments-output")
+    addCommentsColumn();
+
+    const commentsOutput = document.getElementById("comments_body")
       .appendChild(document.createElement("h2")
         .appendChild(document.createTextNode("Comments")).parentElement);
 
-    const cmtDOM = document.createElement("ul");
+    const cmtDOM = document.createElement("div");
+    let quoteDOM;
```
Comment (VivekShankar) [inline portfolio/src/main/webapp/script.js]: Do you need to declare these variables outside the lambda you're passing to `forEach`?

**Review by VivekShankar — COMMENTED** [review]
Really nice work here Mauro - I love how the side bar looks! Just added a few comments here, feel free to ping me with any questions you have!

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (MauroGuerrero20) [reply portfolio/src/main/webapp/index.html:14]: Yes unfortunately. I just added bootstrap div rows and columns to create the side bar.

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (MauroGuerrero20) [reply portfolio/src/main/webapp/script.js]: Is it better to initialized variables in a for loop if the for loop is their scope? 

I always initialized them outside to improve performance.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (MauroGuerrero20) [reply portfolio/src/main/webapp/script.js]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (MauroGuerrero20) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done.

File: portfolio/src/main/java/com/google/sps/containers/Comment.java (hunk 1 above)
Comment (MauroGuerrero20) [reply portfolio/src/main/java/com/google/sps/containers/Comment.java]: Done.

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (VivekShankar) [reply portfolio/src/main/webapp/script.js]: In Javascript, I don't believe there should be any difference in performance (e.g., check out this StackOverflow [link](https://stackoverflow.com/questions/3684923/javascript-variables-declare-outside-or-inside-loop#:~:text=There%20is%20absolutely%20no%20difference,command%20executed%20at%20run%2Dtime.)). 

Yeah, I'd say it's better to initialize variables in a for loop if the for loop is their scope.
See 5.1.3 in the JS style guide (https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations). By declaring local variables using the `let` keyword inside the loop, you'll minimize their scope to just the loop body.

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (VivekShankar) [reply portfolio/src/main/webapp/index.html:14]: Got it, no problem!

**Review by VivekShankar — APPROVED** [review]
LGTM, thanks for addressing the comments here!

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (MauroGuerrero20) [reply portfolio/src/main/webapp/script.js]: Oh ok, I didn't know that about Javascript

**Pull request merged**